### DISCUSSION TITLE
Offline mode?

### DISCUSSION THREAD

**Discussion by D0ctorWh0** [community/community#14095] [Copilot Conversations] [answered]
Github copilot perfectly helps to solve small everyday tasks that involve repeating similar sections of code or small functions that are often found when solving different tasks. But for example, at my work we do not have Internet access. Is it planned to make an offline version of Github Copilot after the release, which will work on local server/user's computer?

**Comment by markiemm** [comment, marked as answer]
Copilot needs internet access to grab your suggestions as processing is done at Github's side

**Reply by JiveyGuy** [reply]
We know, however, it would make sense that you could configure a few dozen GB of disk for caching that has either similar results to your previous completions or a very dumbed down and compressed model that runs on a local server.

**Reply by shotnothing** [reply]
I think you underestimate the amount of VRAM it takes to run even a dumbed down model that has some semblance of being useful, and I can't see how caching results would even work for code completion without a model

**Comment by D0ctorWh0** [comment]
Yes, but I am asking if there are any plans to make copilot version that will run on local server?

**Reply by chirag127** [reply]
you will need a very high end pc to run the model on pc which is rare.

**Reply by danielsobrado** [reply]
Big companies don't allow to send your code over, and they have powerful computers in their data centers. Of course, offline for individuals doesn't make sense. But the big bucks will mostly be made from the corporate world.

**Comment by rslucena** [comment]
It is strange to think that there are development companies, or that are involved with the technological development process, that do not provide access to some type of internet connection. Is it really worth the effort to implement it? *I could be very wrong too, I don't know...

**Reply by mritzmann** [reply]
But a use case would also be, if you don't want to send the code to a third party (here GitHub)... I mean, theoretically `config.php` or `config.ini` files are also sent to GitHub this way.

**Reply by Matt-17** [reply]
For one client I develop via VPN, where external internet connections are not allowed. Would be nice, if it would work there as well. But right now, this is not the biggest issue...

**Reply by good1pen** [reply]
What's the biggest issue?

**Reply by danielsobrado** [reply]
Banks, healthcare etc... they need to comply with audits and regulations, they cannot just let employees connect freely to Internet and risk leaking code and data, is not about the effort to implement it, in fact the effort to prevent it is much bigger.

**Reply by chickdan** [reply]
What about trying to code during a flight? At the park? In another country with slow internet? Just because _you_ don't have a use case for an offline Copilot doesn't mean nobody else does.

**Reply by ewokthegreat** [reply]
I might be late to the party but, if you work on classified/sensitive applications those are unlikely to EVER touch the internet.

**Reply by bobydo** [reply]
I work for a company that doesn't allow developer access intenet because of regulation, so it would be nice to have offline versioin with some cached data in certain categories, for example, .net, python etc..... even small categories, like c# api, pthon robot framework etc. Developers could download them accordingly

**Comment by NeuralAIM** [comment]
Use tabnine for local, and Copilot for online autocomplete : )

**Reply by sisrfeng** [reply]
Without Internert, tabnine, even if you have bought Pro, only has inline completion, but no  full function completion.

**Comment by simonjcarr** [comment]
In some cases copilot is blocked for legal or security reason. As an example I work for a defence company, obviously we can't share our code, but copilot offline would be brilliant.
